### PR TITLE
#72: Add missing methods to AsyncContacthub.java (closes #72)

### LINKED ISSUE #72: Add missing methods to AsyncContacthub.java
[Trello card](https://trello.com/c/58b9456b27211e30f6238fc4/)

### PR DESCRIPTION
**Pull request by dhinus**
Closes #72

### tests performed
* ✅ `sbt test`
* ✅ `sbt example/run`
* ✅ `sbt doc`

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.}

### CODE REVIEW

File: src/main/java/it/contactlab/hub/sdk/java/AsyncContactHub.java [hunk 1]
```diff
@@ -26,7 +37,7 @@ public AsyncContactHub(Auth auth) {
     T apply() throws Exception;
   }
 
-  private <T> CompletionStage<T> wrapAsync(Thunk<T> fun) {
+  private <T> CompletableFuture<T> wrapAsync(Thunk<T> fun) {
```
Comment (gabro) [inline src/main/java/it/contactlab/hub/sdk/java/AsyncContactHub.java]: why are you returning a `CompletableFuture` instead of a `CompletionStage`?

Comment (dhinus) [inline src/main/java/it/contactlab/hub/sdk/java/AsyncContactHub.java]: `CompletableFuture` seems more specific (`CompletableFuture` implements `CompletionStage`). Is there a reason to use `CompletionStage` instead?

Comment (gabro) [inline src/main/java/it/contactlab/hub/sdk/java/AsyncContactHub.java]: As discussed offline, `CompletionStage` represents the read-only side of the computation.
We decided to expose a read-only async API, as it appears to be the best practice both in other languages' SDKs, such as Scala's, both in other Java libraries such as akka (see for instance: http://doc.akka.io/docs/akka/2.4/java/stream/stream-quickstart.html)

File: src/main/java/it/contactlab/hub/sdk/java/internal/api/CustomerApi.java [hunk 2]
```diff
@@ -88,6 +59,49 @@ public static Customer get(Auth auth, String id)
...
+    Function<Integer, CompletionStage<AsyncPaginated<Customer>>>
+        requestFunction = (Integer pageNumber) ->
+            asyncGet(auth, options.withPage(pageNumber));
+
+    return CompletableFuture.supplyAsync(() -> {
+      try {
+        Paged<Customer> pagedCustomers = getPaged(auth, options);
+
+        return new AsyncPaginated<Customer>(pagedCustomers, requestFunction);
+      } catch (Exception ex) {
```
Comment (bytecodeguru) [inline src/main/java/it/contactlab/hub/sdk/java/internal/api/CustomerApi.java]: Is it possible to declare explicitly which exceptions to catch?

File: src/main/java/it/contactlab/hub/sdk/java/AsyncContactHub.java [hunk 3]
```diff
@@ -187,7 +204,7 @@ public String createSessionId() {
    * @param likeId     The id of the Like to be removed.
    * @return           true if the removal was successful.
```
Comment (bytecodeguru) [inline src/main/java/it/contactlab/hub/sdk/java/AsyncContactHub.java]: there seems to be some residual `@return true` in the javadoc

File: src/main/java/it/contactlab/hub/sdk/java/AsyncContactHub.java [hunk 4]
```diff
@@ -267,7 +284,7 @@ public String createSessionId() {
    * @param educationId The id of the Education to be removed.
    * @return            true if the removal was successful.
```
Comment (bytecodeguru) [inline src/main/java/it/contactlab/hub/sdk/java/AsyncContactHub.java]: `@return true`

File: src/main/java/it/contactlab/hub/sdk/java/AsyncContactHub.java [hunk 5]
```diff
@@ -303,7 +320,7 @@ public String createSessionId() {
    * @param newEvent The {@link Event} to create.
    * @return Whether the Event was successfully queued for insertion.
```
Comment (bytecodeguru) [inline src/main/java/it/contactlab/hub/sdk/java/AsyncContactHub.java]: `@return ` (something that evaluates to boolean) :-P

**Pull request merged**